engine/app/components/citizens_advice_components/icons/base.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module CitizensAdviceComponents
+  module Icons
+    class Base < CitizensAdviceComponents::Base
+      attr_reader :size
+
+      def initialize(size: nil, icon_name: nil)
+        super
+        @size = fetch_or_fallback(allowed_values: [nil, :small, :large], given_value: size, fallback: nil)
+        @icon_name = icon_name
```

Comment:
Oh now I can't see where `icon_name` is used at all?

Reply:
Oh wait. It's used in `human_class_name`. Yeah probably makes more sense for the specialised class to define that method itself.

Reply:
For posterity: it's used to render the correct CSS class, eg `cads-icon--arrow-left`.  Ultimately in `css_classes`, via `human_class_name`